modules/consentManagement.js

```diff
@@ -0,0 +1,171 @@
+ * bidRequest object for any supported adapters to read/pass along to their system.
+ * @param {object} config This is the same param that's used in pbjs.requestBids.  The config.adunits will be updated.
+ * @param {function} fn The next function in the chain, used by hook.js
+ */
+export function requestBidsHook(config, fn) {
+  adUnits = config.adUnits || $$PREBID_GLOBAL$$.adUnits;
+  context = this;
+  args = arguments;
+  nextFn = fn;
+  cmpActive = true;
+
+  // in case we already have consent (eg during bid refresh)
+  if (consentId) {
+    applyConsent(adUnits, consentId);
+    return nextFn.apply(context, args);
+  }
+
+  if (!Object.keys(cmpCallMap).includes(userCMP)) {
+    utils.logWarn(`CMP framework (${userCMP}) is not a supported framework.  Aborting consentManagement module and resuming auction.`);
+    return nextFn.apply(context, args);
+  }
+  cmpCallMap[userCMP].call();
+}
+
+// after we have grabbed ideal ID from CMP, apply the data to adUnits object and finish up the module
+function postLookup(consentString) {
+  if (cmpActive) {
+    if (typeof consentString !== 'string' || consentString === '') {
+      exitFailedCMP(`CMP returned unexpected value during lookup process; returned value was (${consentString}).`);
+    } else {
+      clearTimeout(timer);
+      consentId = consentString;
+      applyConsent(adUnits, consentString);
+      nextFn.apply(context, args);
+    }
+  }
+}
+
+function cmpTimedOut() {
+  if (cmpActive) {
+    exitFailedCMP('CMP workflow exceeded timeout threshold.');
+  }
+}
+
+function exitFailedCMP(message) {
+  cmpActive = false;
+  if (lookupFailureChoice === 'proceed') {
+    utils.logWarn(message + ' Resuming auction without consent data as per consentManagement config.');
+    applyConsent(adUnits, undefined);
+    nextFn.apply(context, args);
+  } else {
+    utils.logError(message + ' Canceling auction as per consentManagement config.');
+  }
+}
+
+// assuming we have valid consent ID, apply to adUnits object
+function applyConsent(adUnits, consentString) {
+  adUnits.forEach(adUnit => {
+    adUnit['gdprConsent'] = {
+      consentString: consentString,
+      consentRequired: true
+    };
+  });
+}
+
+/**
+ * Simply resets the module's consentId variable back to its default value.
+*/
+export function resetConsentId() {
+  consentId = '';
+}
+
+/**
+ * A configuration function that initializes some module variables, as well as add a hook into the requestBids function
+ * @param {object} config consentManagement module config settings; cmp (string), waitForConsentTimeout (int), lookUpFailureResolution (string)
+ */
+export function setConfig(config) {
+  if (typeof config.cmp === 'string') {
+    userCMP = config.cmp;
+  } else {
+    userCMP = DEFAULT_CMP;
+    utils.logInfo(`consentManagement config did not specify cmp.  Using system default setting (${userCMP}).`);
+  }
+
+  if (typeof config.waitForConsentTimeout === 'number') {
+    consentTimeout = config.waitForConsentTimeout;
```

Comment:
I think this should just be called `timeout`.  It's already under the `consentManagement` property, so we know it relates to that and the way it's worded here makes the value expected ambiguous (I would guess a `bool` w/o looking at the code)

Reply:
Good point, I changed the name to `timeout`.